vllm_spyre/platform.py

```diff
@@ -111,14 +111,13 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
         if scheduler_config.is_multi_step:
             raise NotImplementedError
 
-        # Can be simplified after the model_config change from vllm:main
+        # Can be simplified after the deprecation of `model_config.task` in
+        # vllm > 0.10.0
         is_decoder = model_config.task == "generate" \
-            if model_config.task \
-                else "generate" in model_config.supported_tasks
+            or "generate" in model_config.supported_tasks
```

Comment:
Won't this condition break if `model_config.task` is `embed` for 0.9.2? Since embed will be in the supported tasks and